test/e2e/storage/subpath.go

```diff
@@ -356,7 +356,7 @@ func testBasicSubpathFile(f *framework.Framework, contents string, pod *v1.Pod,
 func testPodSubpath(f *framework.Framework, subpath, volumeType string, source *v1.VolumeSource) *v1.Pod {
 	var (
 		suffix          = strings.ToLower(fmt.Sprintf("%s-%s", volumeType, rand.String(4)))
-		privileged      = true
+		privileged      = false
```

Comment:
@gnufied fyi

Reply:
also cc @jsafrane 
Not sure if the containerized kubelet tests required privileged pods?

Reply:
thanks. I will run these on RHEL or other SELinux enabled distros. I think these tests should still be able to run without requiring privileged. We might just have to fine tune the test harness.

Reply:
The execution in aws job was not successful.
Went through the logs:
```
I0706 14:48:54.699] Cluster is starting.  It should be ready in a few minutes.
```
And then is unable to connect to it
```
W0706 14:48:54.800] 2018/07/06 14:48:54 process.go:152: Step '/workspace/kops update cluster e2e-96237-dba53.test-cncf-aws.k8s.io --yes' finished in 25.085543268s
W0706 14:48:54.800] 2018/07/06 14:48:54 process.go:150: Running: kubectl get nodes -ojson
W0706 14:48:54.908] Unable to connect to the server: dial tcp: lookup api.e2e-96237-dba53.test-cncf-aws.k8s.io on 10.63.240.10:53: no such host
W0706 14:48:54.911] 2018/07/06 14:48:54 process.go:152: Step 'kubectl get nodes -ojson' finished in 206.555963ms
W0706 14:48:54.912] 2018/07/06 14:48:54 kubernetes.go:34: kubectl get nodes failed: error during kubectl get nodes -ojson: exit status 1
W0706 14:48:54.912] 2018/07/06 14:48:54 kubernetes.go:69: kubectl get nodes failed, sleeping: error during kubectl get nodes -ojson: exit status 1
W0706 14:49:24.912] 2018/07/06 14:49:24 process.go:150: Running: kubectl get nodes -ojson
W0706 14:49:25.173] Unable to connect to the server: dial tcp: lookup api.e2e-96237-dba53.test-cncf-aws.k8s.io on 10.63.240.10:53: no such host
```
It would seem, that this is broader setup issue that happens prior to any test execution.

@gnufied @msau42 What would you suggest as next steps?

Reply:
Aws cluster setup often fails. I would just retry the test

Reply:
Okay, so while ephemeral storage tests do pass on selinux distros without the privileged permissions, rest of the test suite such as hostpath etc will break.

I think, we should may be parameterize `testPodSubpath` so as ephemeral storage(atomic writer) tests can run without privileged pod and other tests in this suite can run as before. Does this work?

Reply:
Do you know what operation the test is doing that requires privileged pods for the persistent volume types?  Maybe we can try to restructure the test so that it doesn't require privileged, if possible.

Reply:
(eventually we'll want the persistent volume tests to also run in a conformance profile suite)

Reply:
/hold
while we figure out what is the best course of action is

Reply:
I haven't dug deeply enough yet.  the hostpath subpath tests are failing for example - https://gist.github.com/gnufied/7b21bff599f24a4897d16ed893777527 will look closely today/tonight

Reply:
For hostpaths - these tests do not pass on selinux enabled distros because `hostPath` volume does not support selinux relabelling - https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/host_path/host_path.go#L209 so we never relabel those volumes.

Error from pod is:

```
─▪ kubectl logs pod-subpath-test-hostpathsymlink-q848 -n e2e-tests-subpath-5hlzl -c init-volume-hostpathsymlink-q848
mkdir: can't create directory '/test-volume/e2e-tests-subpath-5hlzl': Permission denied
```

I would think that other volume types that do support selinux should pass, but I haven't yet tested it on those type of clusters.

Reply:
Thanks, it sounds like the best course of action then is to make privileged a parameter that's false by default, and explicitly set to the true for hostpath volumes.  @mkurylec, can you look into making that change?